modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,56 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
-import eu.timepit.refined.boolean.Not
+import eu.timepit.refined.boolean.Xor
 import eu.timepit.refined.cats._
-import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.Interval.Open
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
-  case class Amount(
-    value: AmountValue,
-    currency: Currency
-  )
-
-  type AmountValueRefined = Long Refined Not[Equal[0L]]
+  // Only allow values in the closed interval 1 to Long.MaxValue and its negation
+  // Since we have to use literal constants here is visual confirmation:
+  //   From Scala Long:       final val MinValue = java.lang.Long.MIN_VALUE
+  //   From java.lang.Long:   @Native public static final long MIN_VALUE = 0x8000000000000000L;
```

Comment:
please use scaladoc style for such comments:

```scala
/** Provides a service as described.
 *
 *  This is further documentation of what we're documenting.
 *  Here are more details about how it works and what it does.
 */
def member: Unit = ()
```

Reply:
Removed the comments now that we're able to use variables.

Reply:
Removed the comments now that we're able to use variables.